File: pp.cs
using System;
using System.Collections.Generic;

class PrimePartitions {

  public bool isPrime(int n) {
      return !divisible(n, n-1);
  }

  public bool divisible(int n, int k) {
      return ((k > 1) && ((n % k == 0) || divisible(n, k - 1))) || (n < 2);
  }

  public List<int> primes(int a, int b) {
    List<int> primes = new List<int>();
    for(int i = a; i <= b; i++) {
      if(isPrime(i)) {
        primes.Add(i);
      }
    }
    return primes;
  }

  public List<List<int>> prime_partitions(int n, int k) {

      List<int> primes_list = new List<int>(primes(k,n));
      List<List<int>> partitions = new List<List<int>>();
      foreach(int i in primes_list) {
          if(i == n) {
              List<int> temp = new List<int>();
              temp.Add(i);
              partitions.Add(temp);
          }
          else {
              foreach(List<int> p in prime_partitions(n - i,  i + 1)) {
                  List<int> temp = new List<int>(p);
                  temp.Insert(0, i);
                  partitions.Add(temp);
              }
          }
      }
      return partitions;
  }

  public static void Main(String[] args) {

      Console.WriteLine("Please enter an integer (preferably under 100):");
      int input = Convert.ToInt32(Console.ReadLine());
      PrimePartitions pp = new PrimePartitions();
      List<List<int>> prime_parts = pp.prime_partitions(input, 1);

      Console.WriteLine("The set of all prime numbers less than " + input + " is: ");

      foreach(List<int> list in prime_parts) {
          for(int i = 0; i < list.Count; i++) {
              Console.Write(list[i]);
              if(i != list.Count - 1) {
                  Console.Write(" + ");
              }
          }
          Console.WriteLine();
      }

  }
}

File: pp.py
import math

def isPrime(N):
    if N <= 1:
        return False
    for i in range(2, int(math.sqrt(N)+1)):
        if N % i == 0:
            return False
    return True

def primes(a, b):
    primes = []
    for i in range(a, b+1):
        if(isPrime(i)):
            primes.append(i)
    return primes

def prime_partitions(n, k):
    primes_list = primes(k,n)
    partitions = [[]]
    for i in primes_list:
        if i == n:
            temp = []
            temp.append(i)
            partitions.append(temp)
        else:
            for p in prime_partitions(n-i,i+1):
                temp = p
                temp.insert(0,i)
                partitions.append(temp) 
    return partitions


value = input("Enter your value: ")
temp = int(value)

total = prime_partitions(temp,1)

finalPartitions = [[]]
for i in range(0,len(total)):
    sum = 0
    for j in range(0,len(total[i])):
        sum += total[i][j]
    if sum == temp:
        finalPartitions.append(total[i])
finalPartitions.pop(0)
for i in range(0,len(finalPartitions)):
    for j in range(0, len(finalPartitions[i])):
        if j == len(finalPartitions[i])-1:
            print(finalPartitions[i][j])
        else:
            print(finalPartitions[i][j], " + ", end='')
